ui/src/main/webapp/src/app/reports/technologies/hibernate/hibernate-report.component.ts

```diff
@@ -0,0 +1,135 @@
+        private _notificationService: NotificationService,
+
+    ) {
+        super(router, activatedRoute, routeFlattener);
+    }
+
+    ngOnInit(): void {
+        this.addSubscription(this.flatRouteLoaded.subscribe(flatRouteData => {
+            this.title = flatRouteData.data.displayName;
+
+            this.loadFilterFromRouteData(flatRouteData);
+
+            const execId = this.execution.id;
+
+            this.techReportService.getHibernateEntityModel(execId).subscribe(
+                data => {
+                    this.entities = data;
+                    this.loading.entities = false;
+                },
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error));
+                });
+
+            this.techReportService.getHibernateConfigurationFileModel(execId).subscribe(configurations => {
+                this.configurationFiles = configurations;
+                this.loading.configurationFiles = false;
+            }, error => {
+                this._notificationService.error(utils.getErrorMessage(error));
+            });
+
+            this.techReportService.getHibernateMappingFileModel(execId).subscribe(mappings => {
+                this.mappingFiles = mappings;
+                this.loading.mappingFiles = false;
+            }, error => {
+                this._notificationService.error(utils.getErrorMessage(error));
+            });
+
+            this.techReportService.getHibernateSessionFactoryModel(execId).subscribe(sessionFactories => {
+                this.sessionFactories = sessionFactories;
+                this.loading.sessionFactories = false;
+            }, error => {
+                this._notificationService.error(utils.getErrorMessage(error));
+            });
+        }));
+    }
+
+    updateSearch() {
+        const regex = new RegExp(this.searchText, 'i');
+
+        this.filterCallbacks = Object.assign({}, {
+            entities: this.filterEntities(regex),
+            mappingFiles: this.filterMappingFiles(regex),
+            configurationFiles: this.filterConfigurationFiles(regex),
+            sessionFactories: this.filterSessionFactories(regex)
+        });
+    }
+
+    clearSearch() {
+        this.searchText = '';
+        this.filterCallbacks = Object.assign({}, this.emptyFilterCallbacks);
+    }
+
+    filterEntities(regex: string|RegExp) {
+        return (entity) => {
+            return entity.resolved.javaClass.qualifiedName.search(regex) !== -1
+                || entity.tableName.search(regex) !== -1;
+        }
+    }
+
+    filterMappingFiles(regex: string|RegExp) {
+        return (file) => {
+            return file.cachedPrettyPath.search(regex) !== -1;
+        }
+    }
+
+    filterConfigurationFiles(regex: string|RegExp) {
+        return this.filterMappingFiles(regex);
+    }
+
+    filterSessionFactories(regex: string|RegExp) {
+        return (factory) => true;
+    }
```

Comment:
We could also create a convenience function that would return a filter, given a list of props as a parameter.

```
let filter: (item: any) => boolean = createFilter("cachedPrettyPath");
let filter: (item: any) => boolean = createFilter(["resolved.javaClass.qualifiedName", "tableName"]);
```
WDYT?
(Not a change request, just a thought)